pkg/eval/math/math.go

```diff
@@ -0,0 +1,286 @@
+// Package math exposes functionality from Go's math package as an elvish
+// module.
+package math
+
+import (
+	"math"
+
+	"github.com/elves/elvish/pkg/eval"
+	"github.com/elves/elvish/pkg/eval/vars"
+)
+
+//elvdoc:var e
+//
+// ```elvish
+// $math:e
+// ```
+//
+// The value of E: 2.718281.... This variable is read-only.
+
+//elvdoc:var pi
+//
+// ```elvish
+// $math:pi
+// ```
+//
+// The value of Pi: 3.141592.... This variable is read-only.
+
+//elvdoc:fn abs
+//
+// ```elvish
+// math:abs $number
+// ```
+//
+// Compute the absolute value of a number. Example:
+//
+// ```elvish-transcript
+// ~> math:abs 1.2
+// ▶ (float64 1.2)
+// ~> math:abs -5.3
+// ▶ (float64 5.3)
+// ```
+
+//elvdoc:fn ceil
+//
+// ```elvish
+// math:ceil $number
+// ```
+//
+// Compute the floor value of a number. Example:
+//
+// ```elvish-transcript
+// ~> math:ceil 1.1
+// ▶ (float64 2)
+// ~> math:ceil -2.3
+// ▶ (float64 -2)
+// ```
+
+//elvdoc:fn cos
+//
+// ```elvish
+// math:cos $number
+// ```
+//
+// Compute the cosine of a number in radians (not degress). Example:
+//
+// ```elvish-transcript
+// ~> math:cos 0
+// ▶ (float64 1)
+// ~> math:cos 3.14159265
+// ▶ (float64 -1)
+// ```
+
+//elvdoc:fn floor
+//
+// ```elvish
+// math:floor $number
+// ```
+//
+// Compute the floor value of a number. Example:
+//
+// ```elvish-transcript
+// ~> math:floor 1.1
+// ▶ (float64 1)
+// ~> math:floor -2.3
+// ▶ (float64 -3)
+// ```
+
+//elvdoc:fn is-inf
+//
+// ```elvish
+// math:is-inf $number $sign
+// ```
```

Comment:
This function could have a more [Elven](https://elv.sh/ref/name.html#related-words) API by taking the `$sign` argument as an option instead, and defaulting to 0. This way it can be used as follows:

```
math:is-inf $n # tests if $n is +inf or -inf
math:is-inf $n &sign=-1 # tests if $n is -inf
math:is-inf $n &sign=1 # tests if $n is +inf
```

I also like how https://godoc.org/math#IsInf explains the use of `$sign` in separate sentences - it's more readable that way.

Reply:
SGTM